weblm/crawler.py

```diff
@@ -77,8 +79,11 @@ def click(self, id):
             y_d += 5 * int(y_d > 0)
 
             self.page.evaluate(f"() => window.scrollTo({x_d}, {y_d})")
-
-            self.page.mouse.click(x - x_d, y - y_d)
+            if x_d or y_d:
+                # not entirely sure this will work if there is scrolling
+                self.page.mouse.click(x - x_d, y - y_d)
+            else:
+                self.page.mouse.click(x + element["origin_x"], y + element["origin_y"])
```

Comment:
(origin_x, origin_y) is the top left corner of the element's box (line 343 below) and (x, y) is the center of the element's box (line 345). Adding them together should throw you very far outside the box, no?

Normally you'd click directly in the center (ie. just (x, y)), but if that center isn't currently in the window we need to scroll it into view which is what I'm doing on line 81.